docs/serverless.rst

```diff
@@ -0,0 +1,121 @@
+Rally and Elastic Serverless
+============================
+
+Overview
+--------
+
+`Elastic Serverless <https://docs.elastic.co/serverless>`_ (later called "Serverless") is a fully managed solution hiding the complexity of Elasticsearch clusters including details such as nodes, shards, data tiers and scaling. Elasticsearch API used in Serverless is a subset of the API available in traditional Elasticsearch installations so Rally had to be modified to support it.
+
+This section explains what was changed in Rally, and what track modifications are required to move existing workloads to Serverless.
+
+.. note::
+
+    Elastic Serverless is in technical preview and is subject to change. Released Rally versions might be lagging behind those changes. If you run into Rally errors when used with Serverless, please try the latest :ref:`development version<dev_installation_instructions>` before reaching out for `help <https://github.com/elastic/rally#getting-help>`_.
+
+Rally
+-----
+
+Rally determines if it is benchmarking a Serverless cluster looking at ``version.build_flavor`` attribute returned by Elasticsearch root endpoint (``/``). If Serverless is confirmed, Rally *automatically* skips all tasks based on :ref:`operations<track_operations>` status in Serverless. For instance, a task with :ref:`node-stats<operation_node_stats>` operation is skipped because ``/_nodes/stats`` Elasticsearch API is not available in Serverless. The status determination is hardcoded into Rally, so if the API status changes in Elasticsearch, Rally needs to be updated too. The same skip logic applies to :doc:`/telemetry`.
+
+.. note::
+
+    Elasticsearch Serverless API is not publicly documented yet. Today, the ultimate documentation is Elasticsearch `source code <https://github.com/elastic/elasticsearch>`_. Please look for ``@ServerlessScope(Scope.PUBLIC)`` annotation to determine if a given REST action is available in Serverless.
+
+The automatic task skip can be overridden through :ref:`run-on-serverless<track_schedule>` property::
+
+  {
+    [..]
+    "schedule": [
+      {
+        "operation": "node-stats",
+        "clients": 1,
+        "run-on-serverless": True
```

Comment:
Using a JSON boolean value is usually preferred over Python, although Rally might already deal with it appropriately:
```suggestion
        "run-on-serverless": true
```